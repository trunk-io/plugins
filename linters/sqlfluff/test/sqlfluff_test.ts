import * as fs from "fs";
import path from "path";
import { linterCheckTest, linterFmtTest, TestCallback } from "tests";

linterCheckTest({linterName: "sqlfluff", namedTestPrefixes: ["basic_check"]});

// Due to sqlfluff's fix subcommand being disabled by default, we need to manually enable it in our test's
// trunk.yaml.
const fmtCallbacks: TestCallback = (driver) => {
    const trunkYamlPath = path.resolve(driver.sandboxPath ?? "", ".trunk/trunk.yaml");
    const currentContents = fs.readFileSync(trunkYamlPath, "utf8");
    const sqlfluffRegex = /- sqlfluff@((\d.?)+)\n/;
    const newContents = currentContents.replace(sqlfluffRegex, "- sqlfluff@$1:\n        commands: [lint, fix]\n");
    fs.writeFileSync(trunkYamlPath, newContents);
};

linterFmtTest({linterName: "sqlfluff", namedTestPrefixes: ["basic_fmt"], preCheck: fmtCallbacks});
