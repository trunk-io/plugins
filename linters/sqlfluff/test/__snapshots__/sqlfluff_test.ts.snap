// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing linter sqlfluff basic_check 1`] = `
{
  "issues": [
    {
      "code": "L010",
      "column": "1",
      "file": "linters/sqlfluff/test/basic_check.in.sql",
      "issueUrl": undefined,
      "level": "LEVEL_HIGH",
      "line": "1",
      "linter": "sqlfluff",
      "message": "Keywords must be consistently upper case.",
      "ranges": undefined,
      "targetType": "sql",
    },
    {
      "code": "L044",
      "column": "1",
      "file": "linters/sqlfluff/test/basic_check.in.sql",
      "issueUrl": undefined,
      "level": "LEVEL_HIGH",
      "line": "1",
      "linter": "sqlfluff",
      "message": "Query produces an unknown number of result columns.",
      "ranges": undefined,
      "targetType": "sql",
    },
    {
      "code": "L036",
      "column": "1",
      "file": "linters/sqlfluff/test/basic_check.in.sql",
      "issueUrl": undefined,
      "level": "LEVEL_HIGH",
      "line": "1",
      "linter": "sqlfluff",
      "message": "Select targets should be on a new line unless there is only one select target.",
      "ranges": undefined,
      "targetType": "sql",
    },
    {
      "code": "L013",
      "column": "12",
      "file": "linters/sqlfluff/test/basic_check.in.sql",
      "issueUrl": undefined,
      "level": "LEVEL_HIGH",
      "line": "1",
      "linter": "sqlfluff",
      "message": "Column expression without alias. Use explicit \`AS\` clause.",
      "ranges": undefined,
      "targetType": "sql",
    },
    {
      "code": "L010",
      "column": "20",
      "file": "linters/sqlfluff/test/basic_check.in.sql",
      "issueUrl": undefined,
      "level": "LEVEL_HIGH",
      "line": "1",
      "linter": "sqlfluff",
      "message": "Keywords must be consistently upper case.",
      "ranges": undefined,
      "targetType": "sql",
    },
    {
      "code": "L039",
      "column": "22",
      "file": "linters/sqlfluff/test/basic_check.in.sql",
      "issueUrl": undefined,
      "level": "LEVEL_HIGH",
      "line": "1",
      "linter": "sqlfluff",
      "message": "Unnecessary whitespace found.",
      "ranges": undefined,
      "targetType": "sql",
    },
    {
      "code": "L014",
      "column": "24",
      "file": "linters/sqlfluff/test/basic_check.in.sql",
      "issueUrl": undefined,
      "level": "LEVEL_HIGH",
      "line": "1",
      "linter": "sqlfluff",
      "message": "Unquoted identifiers must be consistently lower case.",
      "ranges": undefined,
      "targetType": "sql",
    },
    {
      "code": "L039",
      "column": "27",
      "file": "linters/sqlfluff/test/basic_check.in.sql",
      "issueUrl": undefined,
      "level": "LEVEL_HIGH",
      "line": "1",
      "linter": "sqlfluff",
      "message": "Unnecessary whitespace found.",
      "ranges": undefined,
      "targetType": "sql",
    },
    {
      "code": "L010",
      "column": "29",
      "file": "linters/sqlfluff/test/basic_check.in.sql",
      "issueUrl": undefined,
      "level": "LEVEL_HIGH",
      "line": "1",
      "linter": "sqlfluff",
      "message": "Keywords must be consistently upper case.",
      "ranges": undefined,
      "targetType": "sql",
    },
    {
      "code": "L014",
      "column": "34",
      "file": "linters/sqlfluff/test/basic_check.in.sql",
      "issueUrl": undefined,
      "level": "LEVEL_HIGH",
      "line": "1",
      "linter": "sqlfluff",
      "message": "Unquoted identifiers must be consistently lower case.",
      "ranges": undefined,
      "targetType": "sql",
    },
    {
      "code": "L014",
      "column": "43",
      "file": "linters/sqlfluff/test/basic_check.in.sql",
      "issueUrl": undefined,
      "level": "LEVEL_HIGH",
      "line": "1",
      "linter": "sqlfluff",
      "message": "Unquoted identifiers must be consistently lower case.",
      "ranges": undefined,
      "targetType": "sql",
    },
    {
      "code": "L039",
      "column": "7",
      "file": "linters/sqlfluff/test/basic_check.in.sql",
      "issueUrl": undefined,
      "level": "LEVEL_HIGH",
      "line": "1",
      "linter": "sqlfluff",
      "message": "Unnecessary whitespace found.",
      "ranges": undefined,
      "targetType": "sql",
    },
  ],
  "lintActions": [
    {
      "command": "fix",
      "fileGroupName": "sql",
      "linter": "sqlfluff",
      "parser": undefined,
      "paths": [
        "linters/sqlfluff/test/basic_check.in.sql",
      ],
      "report": undefined,
      "upstream": undefined,
      "verb": "TRUNK_VERB_FMT",
    },
    {
      "command": "lint",
      "fileGroupName": "sql",
      "linter": "sqlfluff",
      "parser": undefined,
      "paths": [
        "linters/sqlfluff/test/basic_check.in.sql",
      ],
      "report": undefined,
      "upstream": undefined,
      "verb": "TRUNK_VERB_CHECK",
    },
  ],
  "taskFailures": [],
  "unformattedFiles": [
    {
      "code": undefined,
      "column": "1",
      "file": "linters/sqlfluff/test/basic_check.in.sql",
      "issueUrl": undefined,
      "level": "LEVEL_HIGH",
      "line": "1",
      "linter": "sqlfluff",
      "message": "Incorrect formatting, autoformat by running 'trunk fmt'",
      "ranges": undefined,
      "targetType": undefined,
    },
  ],
}
`;
